pkg/backend/backend.go

```diff
@@ -35,7 +35,7 @@ type backend struct {
 	dbSubscriptions     database.Subscriptions
 
 	aead    encryption.AEAD
-	m       metrics.Interface
+	m       metrics.MetricsEmitter
```

Comment:
I also don't like calling an interface `Interface`, but if it is the only exported interface in a package - IMO, it is ok. It reads well: `metrics.Interface`.

We have quite a few interfaces called like that across the codebase.

I don't mind renaming them, but if we do that - I would like to avoid repetition: `metrics.MetricsEmitter` - says "metrics" twice.

Reply:
And actually, I think we should split this interface into GaugeEmitter and FloatEmitter interfaces. The two different functions are never used together except in a few test files and `pkg/cosmosdb/metrics.go`  for which it would be a trivial change to update. I think that is a side effect of the weird naming.

Reply:
> If you give the package an alias when you import it

Good point.

> or say you import it with `.` which we could do for this kind of package

We only allow `.` imports of `github.com/onsi/ginkgo` and `github.com/onsi/gomega` at the moment for readability of e2e tests:

https://github.com/Azure/ARO-RP/blob/2c2c257b31dc4d2db79e942da138289cddcb2564/hack/validate-imports/validate-imports.go#L18-L26

We should avoid dot imports when possible to avoid confusion.

I think it is not a concern and we can drop duplicating "metrics".